sources/packages/api/src/route-controllers/institution/institution.institutions.controller.ts

```diff
@@ -27,10 +69,77 @@ export class InstitutionInstitutionsController extends BaseController {
   constructor(
     private readonly institutionService: InstitutionService,
     private readonly institutionControllerService: InstitutionControllerService,
+    private readonly userService: UserService,
+    private readonly accountService: BCeIDService,
+    private readonly locationService: InstitutionLocationService,
   ) {
     super();
   }
 
+  /**
+   * Create institution on Institution setup process.
+   * @param payload
+   * @param userToken
+   */
+  @ApiUnprocessableEntityResponse({
+    description: "Institution user already exist",
+  })
+  @Post()
+  async createInstitution(
+    @Body() payload: InstitutionFormAPIInDTO,
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<PrimaryIdentifierAPIOutDTO> {
+    // Check user exists or not
+    const existingUser = await this.userService.getUser(userToken.userName);
+    if (existingUser) {
+      throw new UnprocessableEntityException("Institution User already exists");
+    }
+
+    // Save institution
+    const institution = await this.institutionService.createInstitution(
+      userToken,
+      payload,
+    );
+
+    return {
+      id: institution.id,
+    };
+  }
+
+  /**
+   * Controller method to get all institution users for institution Admin.
+   * @queryParm page, page number if nothing is passed then
+   * DEFAULT_PAGE_NUMBER is taken
+   * @queryParm pageLimit, page size or records per page, if nothing is
+   * passed then DEFAULT_PAGE_LIMIT is taken
+   * @queryParm searchName, user's name keyword to be searched
+   * @queryParm sortField, field to be sorted
+   * @queryParm sortOrder, order to be sorted
+   * @returns All the institution users for the given institution
+   * with total count.
+   */
+  @IsInstitutionAdmin()
+  @Get("user")
+  async getInstitutionUsers(
+    @Query(PaginationParams.SearchCriteria) searchName: string,
+    @Query(PaginationParams.SortField) sortField: string,
+    @UserToken() user: IInstitutionUserToken,
+    @Query(PaginationParams.SortOrder) sortOrder = FieldSortOrder.ASC,
+    @Query(PaginationParams.Page) page = DEFAULT_PAGE_NUMBER,
+    @Query(PaginationParams.PageLimit) pageLimit = DEFAULT_PAGE_LIMIT,
+  ): Promise<PaginatedResults<InstitutionUserAPIOutDTO>> {
+    return this.institutionControllerService.getInstitutionUsers(
+      user.authorizations.institutionId,
+      {
```

Comment:
if we renamed _searchName_ to _searchCriteria_ the entire object can be simplified as below.
```typescript
{ page, pageLimit, searchCriteria, sortField, sortOrder}
```

Reply:
I saw that you gave a thumbs up for this. Was it changed?

Reply:
Please refresh and see again.